crates/formality-prove/src/prove/is_local.rs

```diff
@@ -61,7 +62,7 @@ judgment_fn! {
         (
             // In principle this rule could be removed and preserve soundness,
             // but then we would accept code that is very prone to semver failures.
-            (may_not_be_provable(is_local_trait_ref(decls, env, assumptions, goal)) => ())
+            (may_not_be_provable(&env, |env| is_local_trait_ref(decls, &env, assumptions, goal)) => ())
             --- ("may be added by upstream in a minor release")
             (may_be_remote(decls, env, assumptions, goal) => ())
         )
```

Comment:
this is "could an upstream crate (a crate we know about) add an impl in a non breaking change"